src/__tests__/to-have-value.js

```diff
@@ -0,0 +1,88 @@
+import {render} from './helpers/test-utils'
+
+describe('.toHaveValue', () => {
+  test('handles value of text input', () => {
+    const {queryByTestId} = render(`
+        <input type="text" value="foo" data-testid="with" />
+        <input type="text" value=""  data-testid="empty" />
+        <input type="text" data-testid="without" />
+    `)
+
+    expect(queryByTestId('with')).toHaveValue('foo')
+    expect(queryByTestId('with')).not.toHaveValue('bar')
```

Comment:
Would it make sense to add a test case like this?

```js
expect(queryByTestId('with')).toHaveValue();
```

The corresponding `.not.toHaveValue()` case is already there for `empty`.